### PR TITLE
Update Usage Reports terraform code to support new error handling

### PR DESCRIPTION
**Pull request by manukm-ibm**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates https://github.ibm.com/BSS/metering-deliverables/issues/2243

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccIBMBillingReportSnapshot'

=== RUN   TestAccIBMBillingReportSnapshotBasic
--- PASS: TestAccIBMBillingReportSnapshotBasic (29.62s)
PASS
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/usagereports    35.628s

=== RUN   TestAccIBMBillingReportSnapshotAllArgs
--- PASS: TestAccIBMBillingReportSnapshotAllArgs (19.33s)
PASS
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/usagereports    25.633s

$make testacc TESTARGS="-run=TestAccIBMBillingSnapshotListDataSourceBasic"

=== RUN   TestAccIBMBillingSnapshotListDataSourceBasic
--- PASS: TestAccIBMBillingSnapshotListDataSourceBasic (12.57s)
PASS
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/usagereports    24.533s
...
```

### CODE REVIEW

**Comment by dpopp07** [comment]
@manukm-ibm confirmed that these changes include the new patterns needed for the error toolchain 👍

File: ibm/conns/config.go [hunk 1]
```diff
@@ -3561,7 +3561,11 @@ func RefreshToken(sess *bxsession.Session) error {
 func EnvFallBack(envs []string, defaultValue string) string {
 	for _, k := range envs {
 		if v := os.Getenv(k); v != "" {
-			return v
+			if strings.Contains(v, "https://") {
+				return v
+			} else {
+				return fmt.Sprintf("https://%s/v1", v)
```
Comment (hkantare) [inline ibm/conns/config.go:3567]: why were these changes made?

Comment (manukm-ibm) [reply ibm/conns/config.go:3567]: These changes were from the generated script. If not required, I can remove them

Comment (hkantare) [reply ibm/conns/config.go:3567]: yes remove it

Comment (manukm-ibm) [reply ibm/conns/config.go]: Done

Comment (dpopp07) [reply ibm/conns/config.go]: FYI: I opened up a PR in the generator that fixes this issue to avoid future confusion.

**Review by hkantare — APPROVED** [review]

**Pull request merged**